src/components/AddItems.jsx

```diff
@@ -0,0 +1,102 @@
+import { useCallback, useState } from 'react';
+import { useStateWithStorage } from '../utils';
+import { addItem } from '../api';
+
+const soonDate = 7;
+const kindOfSoonDate = 14;
+const notSoonDate = 30;
+
+export function AddItems() {
+	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(null);
+	const [itemName, setItemName] = useState('');
+
+	const [listPath] = useStateWithStorage('tcl-shopping-list-path', null);
+
+	const handleTextChange = (event) => {
+		return setItemName(event.target.value);
+	};
+
+	const handleChange = (event) => {
+		const numberOfDays = parseInt(event.target.value);
+		setDaysUntilNextPurchase(numberOfDays);
+	};
+
+	const handleSubmit = useCallback(
+		async (event) => {
+			event.preventDefault();
+			if (itemName === '') {
+				alert('Please add an item name.');
+				return;
+			}
+			if (!daysUntilNextPurchase) {
+				alert('Please select an option for date');
+				return;
+			}
+
+			try {
+				await addItem(listPath, {
+					itemName,
+					daysUntilNextPurchase,
+				});
+				alert(
+					'Item was added to the database!',
+					itemName,
+					daysUntilNextPurchase,
+				);
+			} catch (error) {
+				alert(`Item was not added to the database, Error: ${error.message}`);
+			}
+		},
+		[itemName, daysUntilNextPurchase, listPath],
+	);
+
+	return (
+		<div>
+			<form onSubmit={handleSubmit}>
+				<label htmlFor="item-name">
+					Item Name:
+					<br />
+					<input
+						type="text"
+						id="item-name"
+						onChange={handleTextChange}
+						value={itemName}
+					/>
+				</label>
+
+				<br />
+				<label>
+					<input
+						type="radio"
+						value={soonDate}
+						checked={daysUntilNextPurchase === soonDate}
+						onChange={handleChange}
+					/>
+					Soon
+				</label>
+				<br />
+				<label>
+					<input
+						type="radio"
+						value={kindOfSoonDate}
+						checked={daysUntilNextPurchase === kindOfSoonDate}
+						onChange={handleChange}
+					/>
+					Kind of soon
+				</label>
+				<br />
+				<label>
+					<input
+						type="radio"
+						value={notSoonDate}
+						checked={daysUntilNextPurchase === notSoonDate}
+						onChange={handleChange}
+					/>
+					Not soon
+				</label>
+				<br />
+				<button type="submit">Submit</button>
```

Comment:
I wonder if there's an opportunity to disable the button when the input is empty or a selection hasn't been made, instead of relying on alerts! This could enhance the user experience by setting them up for success from the start!

Reply:
i made the input required, let me know whether that was the right way to go!

Reply:
I do see that the input is still logically required, but it might improve the user experience if we remove the logical check for an empty string and instead explicitly mark the input fields as `required`. This way, the browser handles the validation automatically!

Reply:
Oh, the `TextInputElement` component has been abstracted and includes the `required` prop! Got it!